### PR TITLE
HIVE-27346: Getting exception for wildcard (*) search for database and table name

### PR DESCRIPTION
**Pull request by DanielZhu58**
### What changes were proposed in this pull request?
Changed the function when user search database and table name in some certain pattern.

### Why are the changes needed?
To make sure that if * are used in searching for database and table name, there is no exception.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Unit tests.

### CODE REVIEW

**Review by aturoczy — CHANGES_REQUESTED** [review]
Could you please make a test for it? I think this type of behavioral change should cover by tests cases.

File: standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/TestTablesGetExists.java [hunk 1]
```diff
@@ -271,6 +271,20 @@ public void testGetTables() throws Exception {
     Assert.assertEquals("Found functions size", 1, tables.size());
     Assert.assertTrue("Comparing tablenames", tables.contains(testTables[6].getTableName()));
 
+    // Find tables by using the wildcard sign "*"
+    tables = client.getTables(DEFAULT_DATABASE, "*");
```
Comment (dengzhhu653) [inline standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/TestTablesGetExists.java:275]: I guess we need to test searching the tables with database '*', `getTables(DEFAULT_DATABASE, "*")` means the table pattern is '*', it's not reflect the changes.

File: standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/TestDatabases.java [hunk 2]
```diff
@@ -454,6 +454,22 @@ public void testGetDatabases() throws Exception {
     databases = client.getDatabases("*_not_such_database_*");
     Assert.assertEquals("No such databases size", 0, databases.size());
 
+    // Find all the databases by using * as the database names' pattern
+    databases = client.getDatabases("*");
```
Comment (dengzhhu653) [inline standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/TestDatabases.java:458]: We don't make any changes on the `getDatabases` method, these tests look unrelated to the change.

File: standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/TestTablesGetExists.java (hunk 1 above)
Comment (dengzhhu653) [reply standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/TestTablesGetExists.java:275]: How about testing `client.getTables("def*", "*")` and `client.getTables("*no_such_database*", "*")` the first argument for the database is a pattern.

**Review by dengzhhu653 — APPROVED** [review]

**Comment by wecharyu** [comment]
@DanielZhu58 @dengzhhu653 Why do we need this patch? Actually the function definition declares the dbName as the real database name, it does not mean a pattern like tablePattern.
https://github.com/apache/hive/blob/8c90ec0ce576d6319470f7dc4dd27daebb654dec/standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java#L183-L194

**Comment by dengzhhu653** [comment]
@DanielZhu58 do you have any idea of why need to make the `dbName` as a pattern here? do we need to fix the document if possible?
